modules/op-using-pipelines-as-code-with-gitlab.adoc

```diff
@@ -0,0 +1,198 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-pipelines-as-code.adoc
+
+:_content-type: PROCEDURE
+[id="using-pipelines-as-code-with-gitlab_{context}"]
+= Using {pac} with GitLab 
+
+[role="_abstract"]
+If your organization or project uses GitLab as the preferred platform, you can use {pac} for your repository with a webhook on GitLab. 
+
+[discrete]
+.Prerequisites
+
+* Ensure that {pac} is installed on the cluster.
+
+* For authorization, generate a personal access token as the manager of the project or organization on GitLab.
++
+[NOTE]
+====
+* If you want to configure the webhook using the `tkn-pac` CLI, add the `admin:repo_hook` scope to the token.
+
+* If you use a token scoped for a specific project, the token cannot provide API access to a merge request (MR) sent from a forked repository. In such cases, {pac} displays the result of a pipeline as a comment on the MR.
+====
+
+[discrete]
+.Procedure
+
+. Configure the webhook and create a `Repository` custom resource (CR).
+
+** To configure a webhook and create a `Repository` CR _automatically_ using the `tkn pac` CLI tool, use the following example:
++
+[source,terminal]
+----
+$ tkn pac create repo
+----
++
+.Sample interactive output
+[source,terminal]
+----
+? Enter the Git repository url (default: https://gitlab.com/repositories/project):
+? Please enter the namespace where the pipeline should run (default: project-pipelines):
+! Namespace project-pipelines is not found
+? Would you like me to create the namespace project-pipelines? Yes
+✓ Repository repositories-project has been created in project-pipelines namespace
+✓ Setting up GitLab Webhook for Repository https://gitlab.com/repositories/project
+? Please enter the project ID for the repository you want to be configured,
+  project ID refers to an unique ID (e.g. 34405323) shown at the top of your GitLab project : 17103
+👀 I have detected a controller url: https://pipelines-as-code-controller-openshift-pipelines.apps.awscl2.aws.ospqa.com
+? Do you want me to use it? Yes
+? Please enter the secret to configure the webhook for payload validation (default: lFjHIEcaGFlF):  lFjHIEcaGFlF
+ℹ ️You now need to create a GitLab personal access token with `api` scope
+ℹ ️Go to this URL to generate one https://gitlab.com/-/profile/personal_access_tokens, see https://is.gd/rOEo9B for documentation
+? Please enter the GitLab access token:  **************************
+? Please enter your GitLab API URL::  https://gitlab.com
```

Comment:
1. Because this is a suggestion for the change in the actual CLI, it will be reflected in docs only after the change happens from engineering. It can be picked up the doc team as another issue later on.
2. The outputs are copied verbatim from the CLI, based on the upstream docs. This is not a typo in the downstream docs.

Because there is not much to change in the docs for this comment, marking it as resolved for now.

Reply:
@ppitonak 

>Can't it guess the API URL from repository URL? It could at least suggest it as a default value

I think we can do from PAC side

> there are two colons after word URL::, is that the actual output of CLI or just a typo in documentation?

Will check and update 
if its not typo then will fix it

Thanks fr bringing this up